### ISSUE TITLE
GraphQL tracing with Rails 7.0

### ISSUE THREAD

**Issue by liaden** [DataDog/dd-trace-rb#3358]
**Current behaviour**
Rails commands fail due to the error `uninitialized constant MyAppSchema` 

**Expected behaviour**
Application starts and monitors just like it did on Rails 6.1.

**Steps to reproduce**
* Take working 6.1 app with graphql instrumented and attempt to upgrade app

**Environment**

* **ddtrace version:** 1.18.0
* **Configuration block (`Datadog.configure ...`):** `c.tracing.instrument :graphql, schemas: [MyAppSchema]`
* **Ruby version:** 3.1.3
* **Operating system:**
* **Relevant library versions:**

**Comment by TonyCTHsu** [comment]
👋 @liaden , I noticed that [Rails 7 change to Zeitwerk autoloading](https://rubyonrails.org/2021/9/3/autoloading-in-rails-7-get-ready), could this be case? 

The error looks like could be solved by requiring your schema file, have you tried that?

**Comment by liaden** [comment]
@TonyCTHsu Yeah, however, we'd have to go through and load all of the constants underneath `app/graphql` which for our small app is only 15 files, but for our larger monolith is 667 ruby files.

This seems similar to https://github.com/DataDog/dd-trace-rb/issues/2709 and our configuration would align with [Autoloading and Reloading Use Case 3](https://guides.rubyonrails.org/v7.0/autoloading_and_reloading_constants.html#use-case-3-configure-application-classes-for-engines) where we should specify the class as a string that uses `constantize` to resolve it to the class later?

**Comment by TonyCTHsu** [comment]
👋 @liaden , Thanks for bringing this up.

I have looked into the `exception_controller` option from https://github.com/DataDog/dd-trace-rb/issues/2709 . This was solved because I found out that the option is useless. It is deprecated and will be removed. Hence, I have not addressed the zeitwerk autoloading problem yet. 

From [Autoloading and Reloading Use Case 3](https://guides.rubyonrails.org/v7.0/autoloading_and_reloading_constants.html#use-case-3-configure-application-classes-for-engines), passing a string instead of a class seems reasonable to me. However, it failed with `NameError` from `Object.get_const(schema_string)`, because the specified schema class has not yet been loaded. 

I agree that our configuration should work smoother with `zeitwerk`, but it is going to require a lot of fundamental changes. 

Currently, only the specified schemas are being instrumented, however, I think it would be better to instrument all schemas  when no specification. This could help circumvent the trouble for `zeitwerk` autoloading for some cases (I supposed most of our users are instrumenting all the schemas). What do yo think?

For now, the workaround I could provide would be adding this configuration in your Rails app (assuming your schema is sitting under such path). 

```ruby
config.autoload_once_paths << "#{root}/app/graphql"
```
